scripts/countdown_erc721/4-purchase.ts

```diff
@@ -0,0 +1,90 @@
+import { BigNumber, Contract, Signer, ethers } from 'ethers';
+import { LedgerSigner } from '@anders-t/ethers-ledger';
+import { JsonRpcProvider, Log, TransactionReceipt, TransactionResponse } from '@ethersproject/providers';
+import { parsedEnv } from './env.validation';
+
+require('dotenv').config();
+
+/**
+ * Check out the README file
+ */
+
+async function main() {
+  /*
+   * STEP 1: LOAD SENSITIVE INFORMATION SAFELY
+   */
+
+  const privateKey = parsedEnv.PRIVATE_KEY;
+  const providerURL = parsedEnv.CUSTOM_ERC721_PROVIDER_URL;
+  const isHardwareWalletEnabled = parsedEnv.HARDWARE_WALLET_ENABLED;
+
+  const provider: JsonRpcProvider = new JsonRpcProvider(providerURL);
+
+  let deployer: Signer;
+  if (isHardwareWalletEnabled) {
+    deployer = new LedgerSigner(provider, "44'/60'/0'/0/0");
+  } else {
+    deployer = new ethers.Wallet(privateKey, provider);
+  }
+
+  /*
+   * STEP 2: SET HARDCODED VALUES
+   */
+
+  const contractAddress = '0xd323e2c37ca989f5e923a12bcdf10aa3544fa8fa';
+  const quantity = 1;
+  const price = ethers.utils.parseUnits('100', 6);
```

Comment:
@BravoNatalie this should be `ethers.utils.parseUnits('10', 6);` to get 10 million right?